samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlExecutionContext.java

```diff
@@ -72,7 +70,7 @@ public ScalarUdf createInstance(String clazz, String udfName) {
 
     // Configs should be same for all the UDF methods within a UDF. Hence taking the first one.
     Config udfConfig = udfMetadata.get(udfName).get(0).getUdfConfig();
-    ScalarUdf scalarUdf = ReflectionUtils.createInstance(clazz);
+    ScalarUdf scalarUdf = ReflectionUtil.createInstanceOrNull(clazz, ScalarUdf.class, getClass().getClassLoader());
```

Comment:
How do we know which `ClassLoader` to use? I noticed sometimes the `ClassLoader` of the `this` object is used, sometimes the `ClassLoader` of another class is used (like in `SamzaSqlApplicationConfig`) and sometimes a custom `ClassLoader` is used (like `ArrayListOnlyClassLoader`).

Reply:
If the caller wants to use a custom classloader, then the caller will use that.
Otherwise, in general, I think `getClass().getClassLoader()` is preferable, but that is only accessible from outside of static contexts and constructor contexts.
The usage of the `.class` needs to be used when in a static method or a constructor, since `getClass()` is only an instance method.
Do you think it would be useful to doc this somewhere? (e.g. in `ReflectionUtil`)

Reply:
I think it'll be useful, even if it's not part of the official javadoc comments. Another developer searching for example usage might be confused with the various ways the `ClassLoader` is passed in. Thanks!

Reply:
Yep, it's good that you brought the question up. It'll be good to have some more context about this. I added some javadoc in `ReflectionUtil`.
Also, I wanted to correct something I said earlier: in a constructor scope, it is possible to use `getClass().getClassLoader`. It just can't be used in a static scope.